src/harness/sas_testcase.py

```diff
@@ -165,4 +165,23 @@ def assertChannelsContainFrequencyRange(self, channels, frequency_range):
             
     channels.sort(key=lambda ch: (ch['frequencyRange']['highFrequency']), reverse = True)
     self.assertEqual(channels[0]['frequencyRange']['highFrequency'],\
-             frequency_range['highFrequency'])
\ No newline at end of file
+             frequency_range['highFrequency'])
+
+  def TriggerFullActivityDumpAndWaitUntilComplete(self, server_cert, server_key):
+    request_time = datetime.utcnow()
+    self._sas_admin.TriggerFullActivityDump()
+    signal.signal(signal.SIGALRM,
+                  lambda signum, frame:
+                  (_ for _ in ()).throw(Exception('Full Activity Dump Check Timeout')))
+    # Timeout after 2 hours if it's not completed
+    signal.alarm(7200)
+    # Check generation date of full activity dump 
+    while True:
+       dump_message = self._sas.GetFullActivityDump( server_cert, server_key)
+       dump_time = datetime.strptime(dump_message['generationDateTime'],
+                                               '%Y-%m-%dT%H:%M:%SZ')
+       if request_time <= dump_time:
```

Comment:
After running it in a debugger against the fake_sas, here we have a small problem: the request_time as the microsecond precision and the dump_time the second precision, so the first 'request_time <= dump_time' always failed so we always go to the 10 second sleep.

I suggest to replace the line 171 with:
```
   request_time = datetime.utcnow().replace(microsecond=0)
```

Reply:
changes as suggested